app/api/authentication_api.rb

```diff
@@ -246,11 +248,17 @@ class AuthenticationApi < Grape::API
     # Sign out
     #
     desc 'Sign out'
-    delete '/auth/:auth_token' do
-      token = AuthToken.find_by_auth_token(params[:auth_token])
+    params do
+      requires :username, type: String,  desc: 'User username'
+      requires :auth_token, type: String, desc: 'The user\'s temporary auth token'
+    end
+    delete '/auth' do
+      user = User.find_by_username(params[:username])
+      token_with_value = user.auth_tokens 
+      token = token_with_value.select { |token| token.authentication_token == params[:auth_token] }.first
```

Comment:
Move this into the User model as a method `token_for_text` then replace in the 3 locations where you have it now. So this would become:

user = User.find_by_username(params[:username])
token = user.token_for_text params[:auth_token]

and in this case... we need to move the auth token and username to the header as is done in the other calls.